src/utils/os_zephyr.c

```diff
@@ -204,9 +215,54 @@ void *os_memdup(const void *src, size_t len)
 	return r;
 }
 
+void *os_malloc(size_t size)
+{
+    return k_malloc(size);
```

Comment:
So, far WPA supplicant has been treated as an application, so, it used `malloc`, I do agree that for Zephyr this is more of a kernel service, need to think a bit more on this.

Reply:
since this file is an interface layer between hostap and Zephyr it seems like we should use the k_malloc and other Zephyr kernel primitives.

Reply:
Hi @krish2718 , when test the enterprise mode, for example the eap-tls, it will dynamically allocate large memory, and use malloc is not enough and will lead to connection fail. As the zephyr kernel heap is commonly use and have large size, it's more suitable to use.

Reply:
>  and use malloc is not enough and will lead to connection fail

Well, we could increase the heap.

> As the zephyr kernel heap is commonly use and have large size, it's more suitable to use.

intree Zephyr doesn't have much heap usage, but as David says above, if we treat supplicant as core service of Zephyr, then it's fine to move to `k_heap`.